SDL/Quintuplet.cu

```diff
@@ -602,138 +635,394 @@ __device__ bool SDL::passT5RZConstraint(struct SDL::modules& modulesInGPU, struc
+        float residual5_linear = (layer4 <= 6) ? ((z5 - z1) - slope * (rt5 - rt1)) : ((rt5 - rt1) - (z5 - z1)/slope);
+
+        // creating a chi squared type quantity
+        // 0-> PS, 1->2S
+        residual4_linear = (moduleType4 == 0) ? residual4_linear/0.15f : residual4_linear/5.0f;
+        residual5_linear = (moduleType5 == 0) ? residual5_linear/0.15f : residual5_linear/5.0f;
+        residual4_linear = residual4_linear*100;
+        residual5_linear = residual5_linear*100;
+
+        rzChiSquared = 12 * (residual4_linear * residual4_linear + residual5_linear * residual5_linear);
+        return rzChiSquared < 4.677f;
+    }
+//    rzChiSquared = 12*(residual4 * residual4 + residual5 * residual5 + residual_missing * residual_missing);
+
+//    if (isnan(rzChiSquared)) printf("rzChi2: %f, residual2: %f, inner_pt:%f, pseudo_phi: %f, charge: %i, Px:%f, Py:%f, x1:%f, x2:%f, x3:%f, x4:%f, x5:%f, y1:%f, y2:%f, y3:%f, y4:%f, y5:%f, z1:%f, z2:%f, z3:%f, z4:%f, z5:%f, x_center:%f, y_center:%f, slope1c:%f, slope3c:%f\n", rzChiSquared, residual_missing, inner_pt, pseudo_phi, charge, Px, Py, x1, x2, x3, x4, x5, y1, y2, y3, y4, y5, z1, z2, z3, z4, z5, x_center, y_center, slope1c, slope3c);
+
+//    if(layer1 == 7 and layer2 == 8 and layer3 == 9 and layer4 == 10 and layer5 == 11 and rzChiSquared>100){
+//        printf("rt1:%f, rt2:%f, rt3:%f, rt4:%f, rt5:%f\n", rt1, rt2, rt3, rt4, rt5);
+//        printf("x1:%f, x2:%f, x3:%f, x4:%f, x5:%f\n", x1, x2, x3, x4, x5);
+//        printf("y1:%f, y2:%f, y3:%f, y4:%f, y5:%f\n", y1, y2, y3, y4, y5);
+//        printf("z1:%f, z2:%f, z3:%f, z4:%f, z5:%f\n", z1, z2, z3, z4, z5);
+//        printf("rt4_ex:%f, rt5_ex:%f\n", expectrt4, expectrt5);
+//        printf("z4_ex:%f, z5_ex:%f\n", expectz4, expectz5);
+//        printf("residual_missing:%f\n", residual_missing);
+//        printf("Pt:%f, Px:%f, Py:%f, Pz:%f, charge: %i, residual_missing: %f, residual4: %f, residual5:%f, moduleType3:%i\n", Pt, Px, Py, Pz, charge, residual_missing, residual4, residual5, moduleType3);
+//        if (fabs(rzChiSquared-434.901)<0.01) printf("rzChi2: %f, residual2: %f, residual4: %f, residual5:%f, inner_pt:%f, pseudo_phi: %f, charge: %i, Px:%f, Py:%f, x1:%f, x2:%f, x3:%f, x4:%f, x5:%f, y1:%f, y2:%f, y3:%f, y4:%f, y5:%f, z1:%f, z2:%f, z3:%f, z4:%f, z5:%f, x_center:%f, y_center:%f, slope1c:%f, slope3c:%f\n", rzChiSquared, residual_missing, residual4, residual5, inner_pt, pseudo_phi, charge, Px, Py, x1, x2, x3, x4, x5, y1, y2, y3, y4, y5, z1, z2, z3, z4, z5, x_center, y_center, slope1c, slope3c);
+//        printf("residual_missing:%f\n", residual_missing);
+//    }
+
+    // when building T5, apply 99% chi2 cuts as default, and add to pT5 collection. But when adding T5 to TC collections, appy 95% cut to reduce the fake rate
+    TightCutFlag = false;
     //categories!
     if(layer1 == 1 and layer2 == 2 and layer3 == 3)
     {
-        if(layer4 == 4 and layer5 == 5)
+        if(layer4 == 4 and layer5 == 5) //11
```

Comment:
What do all the numbers in the comment mean (here and below)?

Reply:
The numbers is the region number, corresponding with the correct region shown in the slides.

Reply:
This is not clear at all as it is, right? Can you modify so that this is obvious? Also, which slides? I opened the slides here:
https://github.com/SegmentLinking/TrackLooper/pull/281#issuecomment-1527737799
and doesn't look this is included? Or maybe I missed it, which means it should be more obvious?

Reply:
I can put a link in the code comment so it can help people know about the region numbers